angular/src/app/user-area/qr-code-dialog/qr-code-dialog.component.ts

```diff
@@ -37,17 +41,38 @@ export class QrCodeDialogComponent implements OnInit {
   }
 
   public changeStatus(): void {
-    this.twoFactorStatus = !this.twoFactorStatus;
+    if (!this.initialstate && !this.twoFactorStatus) {
+      this.initialstate = true;
+      this.twoFactorStatus = true;
+    }
+    else { this.twoFactorStatus = !this.twoFactorStatus; }
     this.authService.setTwoFactorStatus(this.twoFactorStatus);
+    this.resetQrCode();
     this.userAreaService.changeTwoFactor(this.twoFactorStatus).subscribe(
-      (res: any) => {},
+      (res: any) => {
+        this.userAreaService.pairing().subscribe(
+          (res: any) => {
+            this.loadQrCode();
+          },
+          (err: any) => {
+            this.snackBar.openSnack(err.message, 4000,
+              'red');
+          }
+        )
+      },
       (err: any) => {
         this.snackBar.openSnack(err.message, 4000,
           'red');
```

Comment:
Lines 53-61: The code is almost the same as in loadQrCode() (same subscription to userAreaService.pairing(), same response to the error). I suggest a refactor, something like:
```typescript
this.userAreaService.changeTwoFactor(this.twoFactorStatus).subscribe(
      (res: any) => {
        this.loadQrCode();
      },
      // Error handled by loadQrCode();
    );
```